DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Parts/BagPartGraphSyncer.cs

```diff
@@ -68,11 +69,10 @@ public BagPartGraphSyncer(IServiceProvider serviceProvider)
                 delayedReplaceRelationshipsCommand.SourceIdPropertyName = mergeNodeCommand.IdPropertyName;
                 delayedReplaceRelationshipsCommand.SourceIdPropertyValue = (string?)mergeNodeCommand.Properties[delayedReplaceRelationshipsCommand.SourceIdPropertyName];
 
-                //todo: move this code into graphSyncHelper
-                //todo: what if want different relationships for same contenttype in different bags!
-                string? relationshipType = graphSyncHelper.GraphSyncPartSettings.BagPartContentItemRelationshipType;
-                if (string.IsNullOrEmpty(relationshipType))
-                    relationshipType = await graphSyncHelper.RelationshipType(contentType);
+                var bagContentItemGraphSyncHelper = _serviceProvider.GetRequiredService<IGraphSyncHelper>();
```

Comment:
Can you not use normal DI via constructor for this?

Reply:
We need an `IGraphSyncHelper` for each content item in the bag. We only know how many items are in the bag during the sync process, not at construction time.